### PR TITLE
Avoid delete segment which has been loaded

### PR DESCRIPTION
**Pull request by xige-16**
/kind improvement
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

File: internal/querynode/segment_loader.go [hunk 1]
```diff
@@ -190,7 +196,7 @@ func (loader *segmentLoader) LoadSegment(ctx context.Context, req *querypb.LoadS
 	err = funcutil.ProcessFuncParallel(segmentNum,
 		concurrencyLevel, loadFileFunc, "loadSegmentFunc")
 	if err != nil {
-		segmentGC()
+		segmentGC(false)
```
Comment (yah01) [inline internal/querynode/segment_loader.go:199]: need to handle block list of load segments task

**Review by yah01 — CHANGES_REQUESTED** [review]

**Review by congqixia — COMMENTED** [review]
If `LoadSegment` returns error, the `watchDeltaChannel` and `FromDmLoadDelete` operations will be skipped. This may cause integrity problem.

File: internal/querynode/segment_loader.go [hunk 2]
```diff
@@ -190,7 +196,7 @@ func (loader *segmentLoader) LoadSegment(ctx context.Context, req *querypb.LoadS
...
 		return err
```
Comment (yah01) [inline internal/querynode/segment_loader.go:200]: can't return error here

File: internal/querynode/segment_loader.go [hunk 3]
```diff
@@ -189,9 +195,15 @@ func (loader *segmentLoader) LoadSegment(ctx context.Context, req *querypb.LoadS
...
+	for i, info := range req.Infos {
+		if !loadWithError[i] {
+			loadDoneSegmentIDs = append(loadDoneSegmentIDs, info.SegmentID)
+		}
+	}
 	if err != nil {
-		segmentGC()
-		return err
+		segmentGC(false)
+		return loadDoneSegmentIDs, err
```
Comment (yah01) [inline internal/querynode/segment_loader.go:206]: need to add the loaded segment into meta

**Pull request merged**